js/scorm/wrapper.js

```diff
@@ -181,33 +181,33 @@ class ScormWrapper {
   }
 
   setIncomplete() {
-    this.setValue(this.isSCORM2004() ? 'cmi.completion_status' : 'cmi.core.lesson_status', COMPLETION_STATE.INCOMPLETE.asLowerCase);
-    if (this.commitOnStatusChange && !this.commitOnAnyChange) this.commit();
+    const options = { shouldCommit: this.commitOnStatusChange || this.commitOnAnyChange };
+    this.setValue(this.isSCORM2004() ? 'cmi.completion_status' : 'cmi.core.lesson_status', COMPLETION_STATE.INCOMPLETE.asLowerCase, options);
   }
 
   setCompleted() {
-    this.setValue(this.isSCORM2004() ? 'cmi.completion_status' : 'cmi.core.lesson_status', COMPLETION_STATE.COMPLETED.asLowerCase);
-    if (this.commitOnStatusChange && !this.commitOnAnyChange) this.commit();
+    const options = { shouldCommit: this.commitOnStatusChange || this.commitOnAnyChange };
+    this.setValue(this.isSCORM2004() ? 'cmi.completion_status' : 'cmi.core.lesson_status', COMPLETION_STATE.COMPLETED.asLowerCase, options);
   }
 
   setPassed() {
+    const options = { shouldCommit: this.commitOnStatusChange || this.commitOnAnyChange };
     if (this.isSCORM2004()) {
-      this.setValue('cmi.completion_status', COMPLETION_STATE.COMPLETED.asLowerCase);
-      this.setValue('cmi.success_status', SUCCESS_STATE.PASSED.asLowerCase);
+      this.setValue('cmi.completion_status', COMPLETION_STATE.COMPLETED.asLowerCase, options);
+      this.setValue('cmi.success_status', SUCCESS_STATE.PASSED.asLowerCase, options);
     } else {
```

Comment:
Could you please convert these else blocks to return earlies now that they're not followed by a final line?